src/caffe/util/gpu_memory.cpp

```diff
@@ -0,0 +1,202 @@
+#include <algorithm>
+#include <vector>
+#include "caffe/common.hpp"
+#include "caffe/util/gpu_memory.hpp"
+
+
+#ifndef CPU_ONLY
+#include "cub/cub/util_allocator.cuh"
+#endif
+
+namespace caffe {
+
+#ifndef CPU_ONLY  // CPU-only Caffe.
+  static cub::CachingDeviceAllocator* cubAlloc = 0;
+  vector<gpu_memory::MemInfo> gpu_memory::dev_info_;
+#endif
+
+  gpu_memory::PoolMode gpu_memory::mode_   = gpu_memory::NoPool;
+  bool                 gpu_memory::debug_ = false;
+
+#ifdef CPU_ONLY  // CPU-only Caffe.
+  void gpu_memory::init(const std::vector<int>&, PoolMode, bool) {}
+  void gpu_memory::destroy() {}
+
+  const char* gpu_memory::getPoolName()  {
+    return "No GPU: CPU Only Memory";
+  }
+#else
+
+  void gpu_memory::init(const std::vector<int>& gpus,
+                        PoolMode m, bool debug) {
+    bool debug_env = (getenv("DEBUG_GPU_MEM") != 0);
+    debug_ = debug || debug_env;
+
+    if (gpus.size() <= 0) {
+      // should we report an error here ?
+      m = gpu_memory::NoPool;
+    }
+
+    switch (m) {
+    case CubPool:
+      initMEM(gpus, m);
+      break;
+    default:
+      break;
+    }
+    if (debug)
+      std::cout << "gpu_memory initialized with "
```

Comment:
Can you just make these VLOG(1) instead of having a custom debug variable? Then users can just set --vmodule=gpu_module=1 to get the lines. 

Additionally, can you avoid logging to stdout and log to stderr (which you get with vlog)?